Molio/Source/Data/DataSource/Network/FirebaseService/FirebaseUserService.swift

```diff
@@ -0,0 +1,69 @@
+import FirebaseStorage
+import FirebaseCore
+import FirebaseFirestore
+
+final class FirebaseUserService: UserService {
+    private let storageManager: FirebaseStorageManager
+    private let db: Firestore
+    private let collectionName: String = "users"
+    init(
+        storageManager: FirebaseStorageManager,
+        db: Firestore
+    ) {
+        self.storageManager = storageManager
+        self.db = db
+        
+        // FirebaseApp이 이미 초기화되었는지 확인
+        if FirebaseApp.app() == nil {
+            FirebaseApp.configure()
+        }
+        
+    }
+    
+    func createUser(_ user: MolioUserDTO) async throws {
+        guard let userData = user.toDictionary() else { return }
+        
+        let docRef = getDocumentReference(documentName: user.id)
+        try await docRef.setData(userData)
+    }
+    
+    func readUser(userID: String) async throws -> MolioUserDTO {
+        let docRef = getDocumentReference(documentName: userID)
+        return try await docRef.getDocument(as: MolioUserDTO.self)
+    }
+    
+    func updateUser(_ user: MolioUserDTO) async throws {
+        guard let updateUserData = user.toDictionary() else { return }
+        let docRef = getDocumentReference(documentName: user.id)
+
+        try await withCheckedThrowingContinuation { (continuation: CheckedContinuation<Void, Error>) in
+            docRef.updateData(updateUserData) { error in
+                if let error = error {
+                    continuation.resume(throwing: error)
+                } else {
+                    continuation.resume(returning: ())
+                }
+            }
+        }
+    }
```

Comment:
updateData에 Firebase에서 Swift Concurrency 사용할 수 있도록 만들어 둔 것이 있습니다!!

```suggestion
    func updatePlaylist(newPlaylist: MolioPlaylistDTO) async throws {
        let docRef = getDocumentReference(documentName: newPlaylist.id)
        
        guard let dictionary = newPlaylist.toDictionary() else {
            throw FirestoreError.failedToConvertToDictionary
        }
        
        return try await docRef.updateData(dictionary)
    }
```

이런 식으로 사용하시면 어떨까요?